docs/02 - Contributing/G - End to End Testing with Different Configs.md

```diff
@@ -0,0 +1,103 @@
+2) `anchor-reference-server-config.yaml` - the configuration file for the **Anchor Reference Server**
+3) `assets-test.json` - the assets file to be used by the **Anchor Platform**
+4) `docker-compose-config.override.yaml` - this file gets merged into `docker-compose-configs/docker-compose.base.yaml
+when running docker-compose`. This file contains references to the specific files (the config directory) to be used
+and which tests (end-to-end) are to be run against the setup.
+5) `stellar.toml` - the TOML file thats served from Anchor Platform's `/.well-known/stellar.toml` endpoint
+
+### Adding a New Configuration for Testing
+1) Create a new directory in `integration-test/docker-compose-configs` 
+2) Copy any configuration file(s) you're interested in modifying from 
+`integration-tests/docker-compose-configs/anchor-platform-default-configs` into the new directory **(Note: you only 
+need to copy the config files you're interested in modifying, `anchor-platform-default-configs/*` files will be 
+used otherwise)**
+3) Make whatever specific config changes you'd like to the copied file(s)
+4) Update the `docker-compose-config.override.yaml` file to reference the configuration files in the newly created
+directory.
+
+   a) Replace `<new-config-directory>` with the name of the directory you created for this new configuration.
+
+   b) Replace `<test-to-run1>` and/or `<test-to-run2>` with the name of the test(s) you'd like run against the newly added
+   configuration (refer to: [End to End Tests](/end-to-end-tests/README.md)).
+
+   ```text
+   version: '2'
+   services:
+     anchor-platform-server:
+       volumes:
+         # add mounts for the new config directory
+         - ./<new-config-directory>:/config_override
+       
+       anchor-reference-server:
+         volumes:
+           # add mounts for the new config directory
+           - ./<new-config-directory>:/config_override
+   
+       end-to-end-tests:
+         command: --domain host.docker.internal:8080 --tests <test-to-run1> --tests <test-to-run2>
+   ```
+
+5) At the top of the `docker-compose-config.override.yaml` file, document what this new configuration encompasses
+6) Add this new configuration to the `Makefile` (Note: replace `<name-of-your-new-test>` and 
+`<new-config-directory>`):
+    ```text
+    run-e2e-test-all:
+    	make run-e2e-test-default-config
+    	make run-e2e-test-allowlist
+    	make run-e2e-test-unique-address
+    	make <name-of-your-new-test>
+    
+    <name-of-your-new-test>:
+		$(SUDO) docker-compose -f $(DOCKER_COMPOSE_TEST_BASE_FILE) \
+		-f $(DOCKER_COMPOSE_TEST_CONFIG_DIR)/<new-config-directory>/$(DOCKER_COMPOSE_TEST_OVERRIDE_FILE) rm -f
+
+		$(SUDO) docker-compose --env-file $(DOCKER_COMPOSE_TEST_CONFIG_DIR)/.env \
+		-f $(DOCKER_COMPOSE_TEST_BASE_FILE) \
+		-f $(DOCKER_COMPOSE_TEST_CONFIG_DIR)/<new-config-directory>/$(DOCKER_COMPOSE_TEST_OVERRIDE_FILE) \
+		up --exit-code-from end-to-end-tests
```

Comment:
I think you missed this update: these instructions need to refer to the new approach where you're using `$(call run_tests,<new-config-directory>)`.

Reply:
updated